Content.Server/Station/StationSystem.cs

```diff
@@ -0,0 +1,164 @@
+    private uint _idCounter = 1;
+
+    private Dictionary<StationId, StationInfoData> _stationInfo = new();
+    /// <summary>
+    /// List of stations for the current round.
+    /// </summary>
+    public IReadOnlyDictionary<StationId, StationInfoData> StationInfo => _stationInfo;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<GameRunLevelChangedEvent>(OnRoundEnd);
+    }
+
+    /// <summary>
+    /// Cleans up station info.
+    /// </summary>
+    private void OnRoundEnd(GameRunLevelChangedEvent eventArgs)
+    {
+        if (eventArgs.New == GameRunLevel.PostRound)
+            _stationInfo = new();
+    }
+
+    public class StationInfoData
+    {
+        public readonly string Name;
+
+        /// <summary>
+        /// Job list associated with the game map.
+        /// </summary>
+        public readonly GameMapPrototype MapPrototype;
+
+        /// <summary>
+        /// The round job list.
+        /// </summary>
+        private readonly Dictionary<string, int> _jobList;
+
+        public IReadOnlyDictionary<string, int> JobList => _jobList;
+
+        public StationInfoData(string name, GameMapPrototype mapPrototype, Dictionary<string, int> jobList)
+        {
+            Name = name;
+            MapPrototype = mapPrototype;
+            _jobList = jobList;
+        }
+
+        public bool TryAssignJob(string jobName)
+        {
+            if (_jobList.ContainsKey(jobName))
+            {
+                switch (_jobList[jobName])
+                {
+                    case > 0:
+                        _jobList[jobName]--;
+                        return true;
+                    case -1:
+                        return true;
+                    default:
+                        return false;
+                }
+            }
+            else
+            {
+                return false;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Creates a new station and attaches it to the given grid.
+    /// </summary>
+    /// <param name="mapGrid">grid to attach to</param>
+    /// <param name="mapPrototype">game map prototype of the station</param>
+    /// <param name="gridComponent">optional grid component of the grid.</param>
+    /// <returns>The ID of the resulting station</returns>
+    /// <exception cref="ArgumentException">Thrown when the given entity is not a grid.</exception>
+    public StationId InitialSetupStationGrid(EntityUid mapGrid, GameMapPrototype mapPrototype, IMapGridComponent? gridComponent = null)
+    {
+        if (!Resolve(mapGrid, ref gridComponent))
+            throw new ArgumentException("Tried to initialize a station on a non-grid entity!");
+
+        var jobListDict = mapPrototype.AvailableJobs.ToDictionary(x => x.Key, x => x.Value[1]);
+        var id = AllocateStationInfo();
+
+        _stationInfo[id] = new StationInfoData(mapPrototype.MapName, mapPrototype, jobListDict);
```

Comment:
AFAIK there is currently no way to set the station name without just creating a new map prototype for station. So currently if you want to spawn 5 knightships, they all get the same name and you can't tell them apart on the latejoin gui. Maybe make the station name an optional argument that can be passed by the loadgamemap command? Or some other way of setting the station name?

Reply:
Sounds good to me, I want to add some station management commands but I *absolutely do not* want to delay this PR any further. I'll just make the thing take a name argument.

Reply:
Having a random lore suffix would be nice, e.g. Knightship LV-426 or whatever.

Reply:
>  Having a random lore suffix would be nice, e.g. Knightship LV-426 or whatever.

I agree, and it's on my TODO.